R/assertions.R

```diff
@@ -1418,25 +1423,23 @@ assert_expr_list <- function(arg, # nolint
     ~ is_call(.x) || is_expression(.x)
   )
   invalidargs <- arg[invalids]
+  index <- if_else(names(invalidargs) == "", as.character(which(invalids)),
+    paste0('"', names(invalidargs), '"')
+  )
 
   if (any(invalids)) {
-    argname <- arg_name(substitute(arg))
-    abort(
-      paste0(
-        "All elements of `",
-        argname,
-        "` must be an expression.\n",
-        paste0(
-          argname,
-          "[[",
-          if_else(names(invalidargs) == "", as.character(which(invalids)), names(invalidargs)),
-          "]] = ",
-          map_chr(invalidargs, expr_label),
-          " is of type ",
-          map_chr(invalidargs, typeof),
-          collapse = "\n"
-        )
-      )
+    cli_abort(
+      message = message %||%
+        c("All elements of {.arg {arg_name}} must be an expression.",
+          i = glue_collapse(
+            glue("{{.arg {arg_name}[[{index}]]}} =
+            {{.code {invalidargs}}} is of type
+                 {{.cls {map_chr(invalidargs, typeof)}}}"),
+            sep = "\n\n"
```

Comment:
Any idea how to start a new line with cli? I used "\n\n" because "\n" only adds a blank
```r
> cli_abort(c("message", i="1\n2"))
Error:
! message
ℹ 1 2
```

Reply:
How come you want a part of the information to appear on the next line?

Reply:
In similar messaging, we've used `sep = ", "` and `last = ", and "`

Reply:
> How come you want a part of the information to appear on the next line?

See the following example, before my updates, error messages start with a new line for each element -  
```r
> example_fun(exprs("AE", DTSEQ = AESEQ, !!list("a"), !!list("a"), !!list("a")))
Error in `assert_expr_list()`:
! All elements of `vars` must be an expression.
vars[[3]] = `list("a")` is of type list
vars[[4]] = `list("a")` is of type list
vars[[5]] = `list("a")` is of type list
Run `rlang::last_trace()` to see where the error occurred.
```
> In similar messaging, we've used `sep = ", "` and `last = ", and "`

I updated it to `sep = ", "` and `last = ", and "` to align with similar messaging. Thanks both!
```r
> example_fun(exprs("AE", DTSEQ = AESEQ, !!list("a"), !!list("a"), !!list("a")))
Error in `example_fun()`:
! All elements of `vars` must be an expression.
ℹ `vars[[3]]` = `list("a")` is of type <list>, `vars[[4]]` = `list("a")` is of type <list>, and `vars[[5]]` = `list("a")` is
  of type <list>
```

Reply:
whew!!  that looks super nice and clean!!  Thanks for paying attention to those details!!

Reply:
Could your example to me be included in the function documentation.  That would be nice to see in our documentation.

Reply:
updated